WordPress/Classes/Services/MediaThumbnailCoordinator.swift

```diff
@@ -25,13 +25,30 @@ class MediaThumbnailCoordinator: NSObject {
         return mediaThumbnailService
     }()
 
+    private lazy var mediaService: MediaService = {
+        let mediaService = MediaService(managedObjectContext: backgroundContext)
+        return mediaService
+    }()
+
     /// Tries to generate a thumbnail for the specified media object with the size requested
     ///
     /// - Parameters:
     ///   - media: the media object to generate the thumbail representation
     ///   - size: the size of the thumbnail
     ///   - onCompletion: a block that is invoked when the thumbnail generatio is completed with success or failure.
     @objc func thumbnail(for media: Media, with size: CGSize, onCompletion: @escaping ThumbnailBlock) {
+        if media.remoteStatus == .stub {
+            guard let mediaID = media.mediaID else {
```

Comment:
Could we extract the body of this `if` into another method? `fetchThumbnailForMediaStub` or something, and pass it the media, size, completion block?